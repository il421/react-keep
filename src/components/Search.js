import React from 'react';

export const Search = () => {

  return (
    <>
      <input className="search" placeholder="Search"/>
    </>
  )
}

export default Search;
