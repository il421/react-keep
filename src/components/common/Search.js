import React from 'react';
import { connect } from 'react-redux';
import { setSearchFilter } from '../../actions/filters';

export const Search = ({ setSearchFilter }) => {

  return (
    <>
      <input
        className="search"
        placeholder="Search"
        onInput={ (evt) => setSearchFilter(evt.target.value) }
      />
    </>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    setSearchFilter: (value) => dispatch(setSearchFilter(value))
  };
};

export default connect(undefined, mapDispatchToProps)(Search);
