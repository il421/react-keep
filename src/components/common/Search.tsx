import React from "react";
import { connect } from "react-redux";
import { setSearchFilter } from "../../actions/filters";
import { Placeholders } from "../../common";
import "../../styles/components/header/_search.scss";
import { Dispatch } from "redux";
import { SearchFilterAction } from "../../store/store.types";

interface SearchProps {
  wrapperClass?: string;
}

interface DispatchProps {
  setSearchFilter: (value: string) => void;
}

type Props = SearchProps & DispatchProps;

class Search extends React.PureComponent<Props> {
  render() {
    return (
      <div className={this.props.wrapperClass}>
        <input
          className="search"
          placeholder={Placeholders.search}
          onInput={evt =>
            this.props.setSearchFilter((evt.target as HTMLInputElement).value)
          }
        />
      </div>
    );
  }
}

const mapDispatchToProps = (
  dispatch: Dispatch<SearchFilterAction>
): DispatchProps => {
  return {
    setSearchFilter: (value: string) => dispatch(setSearchFilter(value))
  };
};

export default connect<{}, DispatchProps, SearchProps>(
  undefined,
  mapDispatchToProps
)(Search);
