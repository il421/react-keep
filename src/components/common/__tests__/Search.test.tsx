import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { DispatchProps, Search } from "../Search";

let wrapper: ShallowWrapper<DispatchProps, any>, props: DispatchProps;

beforeEach(() => {
  props = {
    setSearchFilter: jest.fn(),
  };
  wrapper = shallow<DispatchProps>(<Search {...props} />);
});

test("should render Search", () => {
  expect(wrapper.exists(".search")).toBeTruthy();
  expect(wrapper.debug()).toMatchSnapshot();
});

test("should setSearchFilter when search", () => {
  wrapper.find(".search").simulate("change", {
    target: { value: "test" },
  });
  expect(props.setSearchFilter).toHaveBeenLastCalledWith("test");
});
