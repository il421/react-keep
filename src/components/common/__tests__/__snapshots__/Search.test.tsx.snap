// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search 1`] = `
"<div className={[undefined]}>
  <input className=\\"search\\" placeholder=\\"Search\\" onChange={[Function: onChange]} />
</div>"
`;
